docs/http_api.md

```diff
@@ -0,0 +1,114 @@
+
+The HTTP API is stateless, which means that interactive transactions are not possible. Since all batches are executed as part of transactions, any transaction statements (e.g `BEGIN`, `END`, `ROLLBACK`...) are forbidden and will yield an error.
+
+##### Body
+
+The body for the query request has the following format:
+
+```
+type QueryBody = {
+    statements: Array<Query>
+}
+
+type Query = string | ParamQuery;
+type ParamQuery = { q: string, params: null | Record<string, Value> | Array<Value> }
+```
+
+Queries are either simple strings or `ParamQuery` that accept parameter bindings. The `statements` arrays can contain a mix of the two types.
+
+##### Response Format
+
+On success, a request to `POST /query` returns a response with an HTTP 200 code and a JSON body with the following structure:
+```
+type BatchResponse = {
+    results: Array<QueryResult> | undefined,
+}
+
+type QueryResult = {
+    columns: Array<string>,
+    rows: Array<Array<Value>>,
+}
+
+```
+
+On success, `BatchResponse` `error` is undefined, and `results` are populated. This is accompanied by an HTTP 200 code.
+On error, `results` is undefined, and `error` is populated. This is accompanied by the relevant HTTP error code.
+
+Each entry in the `results` array of the `BatchResponse` corresponds to a query in the request.
+The `QueryResult` is either an error or a set of results.
+
+The `Query` can either be a plain query string, such as `SELECT * FROM users` or `INSERT INTO users VALUES ("adhoc")`, or objects for queries with bound parameters.
+
+##### Parameter binding
+
+Queries with bound parameters come in two types:
+
+1. Named bound parameters, where the parameter is referred to by a name and is prefixed with a `:` or a `$`. If the query uses named parameters, then the `params` field of the query should be an object mapping parameters to their value.
+
+- Example: a query with named bound parameters
+
+```json
+{
+    "q": "SELECT * FROM users WHERE name = :name",
+    "params": {
+        ":name": "adhoc",
+    }
+}
+```
+The prefix of the parameter must be specified in the `params` field (i.e, `:name` instead of `name`), since libSQL threat the two as different parameters.
+
+2. Positional query parameters, bound by their position in the parameter list, and prefixed `?`. If the query uses positional parameters, the values should be provided as an array to the `params` field.
+
+- Example: a query with positional bound parameters
+
+```json
+{
+    "q": "SELECT * FROM users WHERE name = ?",
+    "params": ["adhoc"]
+}
+```
+
+### Error handling
+
+On error, the sqld API returns a unified error payload, with a relevant HTTP status code.
+The error response payload has the following format:
+
+```
+type Error = {
+    error: {
+        message: string,
+        error_code: string,
+   }
+}
+```
+
+The error code can later be used to link to the relevant documentation.
```

Comment:
Ideally the message can contain a URL with the error code baked into it (fragment) to let the developer go straight there.  We just need to commit to way to host these docs.

Reply:
Yes this is the end goal, but first the codes, then we can add the url field when we have somewhere to take the user to